homeassistant/components/matrix/__init__.py

```diff
@@ -368,6 +370,28 @@ def _send_image(self, img, target_rooms):
                     ex.content,
                 )
 
+    def _send_video(self, vid, target_rooms):
+        _LOGGER.debug("Uploading file from path, %s", vid)
+
+        if not self.hass.config.is_allowed_path(vid):
+            _LOGGER.error("Path not allowed: %s", vid)
+            return
+        with open(vid, "rb") as upfile:
+            vidfile = upfile.read()
```

Comment:
Video files can be big, we should not read the whole video into memory but stream upload it.

Reply:
The underlying library uses `requests`, so passing the file instead of the file contents also works here.

@joshgordon See this diff: https://github.com/jurriaan/home-assistant/commit/04533169a26eadc423609685812dedeecfb6aa3c

@balloob Would this change make the file upload logic a bit more acceptable so we can get this merged?

Reply:
Hi @jurriaan! My PR [here](https://github.com/home-assistant/core/pull/72797) is aiming to switch to a new dependency library for `matrix` that uses `aiohttp`; I'm hopeful that if I'm able to get that merged, various features such as video uploading should become a lot more palatable/forward-compatible